CRM/Core/Payment.php

```diff
@@ -518,9 +519,23 @@ public function getText($context, $params) {
             $gotText .= ' ' . ts('You will receive an email receipt for each recurring contribution.');
           }
         }
-        break;
+        return $gotText;
+
+      case 'contributionPageContinueText':
+        if ($params['amount'] <= 0) {
+          return ts('To complete this transaction, click the <strong>Continue</strong> button below.');
+        }
+        if ($this->_paymentProcessor['billing_mode'] == 4) {
```

Comment:
I always get confused with "billing_mode" etc.  Do we have a constant somewhere?  And/or a comment reminding what it means

Reply:
right - it's deprecated but it came across from where I pulled this from. The options are 1 and 4 and ?? maybe 3 not sure.